Index: kate/plugins/backtracebrowser/katebacktracebrowserplugin.desktop
===================================================================
--- kate/plugins/backtracebrowser/katebacktracebrowserplugin.desktop	(.../tags/KDE/4.2.2/kdesdk)	(wersja 956244)
+++ kate/plugins/backtracebrowser/katebacktracebrowserplugin.desktop	(.../branches/KDE/4.2/kdesdk)	(wersja 956244)
@@ -12,6 +12,7 @@
 Name[et]=Kate tagasijälituse brauser
 Name[fr]=Explorateur de piles d'appels de Kate
 Name[gl]=Navegador de trazados inversos para Kate
+Name[hu]=Kate visszakövetési böngésző
 Name[it]=Browser di backtrace di Kate
 Name[ja]=Kate バックトレースブラウザ
 Name[km]=កម្មវិធី​រុករក​ដាន​ថយក្រោយ​របស់ Kate
@@ -38,6 +39,7 @@
 Comment[et]=Tagasijälituse liikumistööriista vaade
 Comment[fr]=Vue des outils de navigation de piles d'appels
 Comment[gl]=Vista da ferramenta de navegación por trazados inversos
+Comment[hu]=Eszköznézet visszakövetéshez
 Comment[it]=Vista degli strumenti di navigazione di backtrace
 Comment[ja]=バックトレースを調べるためのツールビュー
 Comment[km]=ទិដ្ឋភាព​ឧបករណ៍​រុករក​ដាន​ថយក្រោយ
Index: kate/plugins/kate-ctags/katectagsplugin.desktop
===================================================================
--- kate/plugins/kate-ctags/katectagsplugin.desktop	(.../tags/KDE/4.2.2/kdesdk)	(wersja 956244)
+++ kate/plugins/kate-ctags/katectagsplugin.desktop	(.../branches/KDE/4.2/kdesdk)	(wersja 956244)
@@ -13,6 +13,7 @@
 Name[fr]=Module externe pour CTags
 Name[ga]=Breiseán CTags
 Name[gl]=Extensión de CTags
+Name[hu]=CTags-bővítmény
 Name[it]=Estensione CTags
 Name[ja]=CTags プラグイン
 Name[km]=កម្មវិធី​ជំនួយ CTags
@@ -42,6 +43,7 @@
 Comment[et]=Definitsiooni/deklaratsiooni otsimine CTagsiga
 Comment[fr]=Définitions/déclarations de recherche avec CTags
 Comment[gl]=Procura de definicións/declaracións con CTags
+Comment[hu]=Definíciók/deklarációk kikeresése CTags segítségével
 Comment[it]=Cerca definizioni e dichiarazioni con CTags
 Comment[ja]=CTags を使って定義/宣言を検索します
 Comment[km]=រក​មើល​និយមន័យ/កា​រប្រកាស​ជា​មួយ CTags
Index: kate/plugins/kjswrapper/samples/katekjsconsolewindow.desktop
===================================================================
--- kate/plugins/kjswrapper/samples/katekjsconsolewindow.desktop	(.../tags/KDE/4.2.2/kdesdk)	(wersja 956244)
+++ kate/plugins/kjswrapper/samples/katekjsconsolewindow.desktop	(.../branches/KDE/4.2/kdesdk)	(wersja 956244)
@@ -13,7 +13,7 @@
 Name[et]=Kate JavaScripti konsooli aken
 Name[ga]=Fuinneog Chonsóil Javascript Kate
 Name[gl]=Fiestra da consola de Javascript para Kate
-Name[hu]=Kate Javascript konzolablak
+Name[hu]=Kate Javascript-konzolablak
 Name[it]=Finestra di console JavaScript di Kate
 Name[ja]=Kate JavaScript コンソールウィンドウ
 Name[km]=បង្អួច​កុងសូល Javascript Kate
Index: kate/plugins/katebuild-plugin/katebuildplugin.desktop
===================================================================
--- kate/plugins/katebuild-plugin/katebuildplugin.desktop	(.../tags/KDE/4.2.2/kdesdk)	(wersja 956244)
+++ kate/plugins/katebuild-plugin/katebuildplugin.desktop	(.../branches/KDE/4.2/kdesdk)	(wersja 956244)
@@ -12,6 +12,7 @@
 Name[et]=Kate ehutamisplugin
 Name[fr]=Module externe de construction pour Kate
 Name[gl]=Extensión de construción para Kate
+Name[hu]=Lefordító bővítmény a Kate-hez
 Name[it]=Estensione di generazione di Kate
 Name[ja]=Kate ビルドプラグイン
 Name[km]=កម្មវិធី​ជំនួយ​ស្ថាបនា Kate​
@@ -41,6 +42,7 @@
 Comment[fr]=Compiler ou lancer Make et analyser syntaxiquement les messages d'erreur
 Comment[ga]=Tiomsaigh nó Tóg agus parsáil teachtaireachtaí earráide
 Comment[gl]=Compila ou Make e procesa as mensaxes de erro
+Comment[hu]=Fordítás vagy Make művelet a hibaüzenetek feldolgozásával
 Comment[it]=Compila o esegui make ed elabora i messaggi di errore
 Comment[ja]=コンパイルまたは make してエラーメッセージを解析します
 Comment[km]=Compile ឬ Make និង​ញែកសារ​កំហុស
Index: kapptemplate/templates/C++/runner/%{APPNAMELC}.desktop
===================================================================
--- kapptemplate/templates/C++/runner/%{APPNAMELC}.desktop	(.../tags/KDE/4.2.2/kdesdk)	(wersja 956244)
+++ kapptemplate/templates/C++/runner/%{APPNAMELC}.desktop	(.../branches/KDE/4.2/kdesdk)	(wersja 956244)
@@ -10,6 +10,7 @@
 Comment[fr]=Lanceur %{APPNAME}
 Comment[ga]=Feidhmitheoir %{APPNAME}
 Comment[gl]=Executor %{APPNAME}
+Comment[hu]=%{APPNAME}-futtató
 Comment[it]=Esecutore di %{APPNAME}
 Comment[km]=កម្មវិធី​រត់ %{APPNAME}​
 Comment[ko]=%{APPNAME} 실행기
Index: kapptemplate/templates/C++/runner/runner.kdevtemplate
===================================================================
--- kapptemplate/templates/C++/runner/runner.kdevtemplate	(.../tags/KDE/4.2.2/kdesdk)	(wersja 956244)
+++ kapptemplate/templates/C++/runner/runner.kdevtemplate	(.../branches/KDE/4.2/kdesdk)	(wersja 956244)
@@ -9,6 +9,7 @@
 Name[fr]=Modèle de lanceur pour Plasma
 Name[ga]=Teimpléad d'Fheidhmitheoirí Plasma
 Name[gl]=Modelo de executor de Plasma
+Name[hu]=Sablon Plasma-futtató elemhez
 Name[it]=Modello di esecutore Plasma
 Name[ja]=Plasma Runner 用テンプレート
 Name[km]=ពុម្ព​កម្មវិធី​រត់​របស់ប្លាស្មា
@@ -38,6 +39,7 @@
 Comment[fr]=Un modèle de lanceur pour Plasma
 Comment[ga]=Teimpléad d'fheidhmitheoirí Plasma
 Comment[gl]=Un modelo de executor de Plasma
+Comment[hu]=Sablon Plasma-futtatóhoz
 Comment[it]=Modello di esecutore Plasma
 Comment[ja]=Plasma Runner 用のテンプレート
 Comment[km]=ពុម្ពកម្មវិធី​រត់​ប្លាស្មា
Index: doc/scripts/kdesvn-build/kdesvn-build.desktop
===================================================================
--- doc/scripts/kdesvn-build/kdesvn-build.desktop	(.../tags/KDE/4.2.2/kdesdk)	(wersja 956244)
+++ doc/scripts/kdesvn-build/kdesvn-build.desktop	(.../branches/KDE/4.2/kdesdk)	(wersja 956244)
@@ -6,8 +6,10 @@
 Name[de]=KDE-SVN-Build
 Name[es]=Construcción SVN de KDE
 Name[fr]=Compilation de SVN pour KDE
+Name[hu]=KDE-fordítás SVN-ből
 Name[is]=KDE SVN-útgáfa (SVN Build)
 Name[it]=Generazione di KDE da SVN
+Name[lv]=KDE SVN būvējums
 Name[pl]=Budowanie KDE z SVN-a
 Name[pt]=Compilação do SVN do KDE
 Name[pt_BR]=Construtor SVN do KDE
Index: lokalize/src/mergemode/mergecatalog.cpp
===================================================================
--- lokalize/src/mergemode/mergecatalog.cpp	(.../tags/KDE/4.2.2/kdesdk)	(wersja 956244)
+++ lokalize/src/mergemode/mergecatalog.cpp	(.../branches/KDE/4.2/kdesdk)	(wersja 956244)
@@ -146,6 +146,11 @@
     return item;
 }
 
+static QString strip(QString source)
+{
+    source.remove('\n');
+    return source;
+}
 
 bool MergeCatalog::loadFromUrl(const KUrl& url)
 {
@@ -167,31 +172,28 @@
     QMultiHash<QString, int> mergeMap;
     while (i.entry<mergeSize)
     {
-        mergeMap.insert(mergeStorage.source(i),i.entry);
+        mergeMap.insert(strip(mergeStorage.source(i)),i.entry);
         ++(i.entry);
     }
 
     i.entry=0;
     while (i.entry<size)
     {
-        QList<int> entries=mergeMap.values(baseStorage.source(i));
+        QList<int> entries=mergeMap.values(strip(baseStorage.source(i)));
         QList<MatchItem> scores;
         int k=entries.size();
-        if (!k)
+        if (k)
         {
-            ++i.entry;
-            continue;
-        }
-        while(--k>=0)
-            scores<<calcMatchItem(i,DocPosition( entries.at(k) ));
+            while(--k>=0)
+                scores<<calcMatchItem(i,DocPosition( entries.at(k) ));
 
-        qSort(scores);
+            qSort(scores);
 
-        m_map[i.entry]=scores.first().mergeEntry;
+            m_map[i.entry]=scores.first().mergeEntry;
 
-        if (scores.first().translationIsDifferent)
-            m_mergeDiffIndex.append(i.entry);
-
+            if (scores.first().translationIsDifferent)
+                m_mergeDiffIndex.append(i.entry);
+        }
         ++i.entry;
     }
     return true;
Index: lokalize/src/project/project.h
===================================================================
--- lokalize/src/project/project.h	(.../tags/KDE/4.2.2/kdesdk)	(wersja 956244)
+++ lokalize/src/project/project.h	(.../branches/KDE/4.2/kdesdk)	(wersja 956244)
@@ -122,8 +122,8 @@
     void defineNewTerm(QString en=QString(),QString target=QString());
 
 
-    void registerEditor(EditorWindow* e){m_editors<<e;}
-    void unregisterEditor(EditorWindow* e){m_editors.remove(m_editors.indexOf(e));}
+    void registerEditor(EditorWindow* e){}
+    void unregisterEditor(EditorWindow* e){}
 
 
 private:

Zmiany atrybutów dla: .
___________________________________________________________________
Dodane: svn:externals
   + 


